torch/nn/modules/_functions.py

```diff
@@ -36,6 +33,10 @@ def forward(self, input, weight, bias, running_mean, running_var, eps, momentum,
         mean_all_reduce.wait()
         invstd_all_reduce.wait()
 
+        size = count_all.view(-1).long().sum()
+        if size == 1:
+            raise ValueError('Expected more than 1 value per channel when training, got input size {}'.format(size))
+
```

Comment:
cc @ngimel @jjsjann123 @ptrblck 

I found that this change to `size` calculation introduces the huge regression in NCCL reduction. For a sync BN forward reduction of (64, 2048, 4, 4) float tensor on my machine with 2 GPUs, the previous code takes 0.7ms, but current code takes 2.4ms. 

I think this is because the old `size` calculation is purely on CPU, but the new `size` calculation is on GPU, and needs a synchronize for the `size == 1` comparison on CPU. This can be easily recovered by 1. using the old `size` calculation, or 2. remove the `if` statement. 

I'm currently working on migrating the sync BN channels-last from apex, and we can discuss a fix there.